upup/pkg/fi/executor.go

```diff
@@ -171,39 +173,85 @@ func (e *executor[T]) RunTasks(taskMap map[string]Task[T]) error {
 	return nil
 }
 
+// getExecutorPoolSize determines the executor pool size for executing tasks in parallel
+//
+// It will determine the pool size based on the current number of tasks and the maximum pool size.
+// If the maximum pool size is not specified or greater than the actual number of tasks, then the function returns the number of tasks as the pool size.
+// Otherwise it will return the defined maximum pool size.
+//
+// Currently, the max pool size can be set by specifying the ENV variable "KOPS_EXECUTOR_POOL_MAX_SIZE".
+func (e *executor[T]) getExecutorPoolSize(numberOfTasks int) int {
+	// TODO: make configurable via flag rather than env var?
+	executorPoolMaxSize, _ := strconv.Atoi(os.Getenv("KOPS_EXECUTOR_POOL_MAX_SIZE"))
+
+	// if executorPoolMaxSize not specified, or the actual number of tasks being less than the wanted number of workers,
+	// set the number of workers to the number of tasks, thus also - more or less - preserving the old behavior
+	if executorPoolMaxSize <= 0 || executorPoolMaxSize > numberOfTasks {
+		return numberOfTasks
```

Comment:
Not sure anymore, I think it produces a deadlock because of how channels work (or I confuse it with something different I did which actually produced one), and it also did not make sense to me to spawn more goroutines/workers than necessary.